### ISSUE TITLE
Support for RGB camera in acquisition events

### ISSUE THREAD

**Issue by binli123** [micro-manager/pycro-manager#93]
Acquisition event does not support RGB cameras. Currently, the way to use a color camera is via core.snap_image().

**Comment by henrypinkard** [comment]
This will require changes to [NDTiffStorage](https://github.com/micro-manager/NDTiffStorage), [AcqEngJ](https://github.com/micro-manager/AcqEngJ), and [NDViewer](https://github.com/micro-manager/NDViewer) to make sure RGB support is working

**Comment by henrypinkard** [comment]
Added support for this (https://github.com/micro-manager/pycro-manager/pull/117). It will be available in micro-manager nightly builds once this PR merges: https://github.com/micro-manager/micro-manager/pull/943

Good chance that there are more RGB-related things that need fixes too. 

Is your camera 32-bit RGB or 64-bit RGB?

**Comment by binli123** [comment]
> Added support for this (#117). It will be available in micro-manager nightly builds once this PR merges: [micro-manager/micro-manager#943](https://github.com/micro-manager/micro-manager/pull/943)
> 
> Good chance that there are more RGB-related things that need fixes too.
> 
> Is your camera 32-bit RGB or 64-bit RGB?

Glad to hear! It is a 32-bit RGB. Two extra things I noticed regarding color camera in MM2: 1) the image channel order seemed to be reversed in MDA image display window. 2) When I tried to stitch the color image tiles in FIJI I encountered metadata error saying that images should have the same number of channels. I think this might be related to a bug that @marktsuchida fixed in MM2 beta but might still exist in MM2 gamma?

**Comment by henrypinkard** [comment]
For 1), you mean using an MDA that you set up i the GUI in micro-manager, or one that you run from pycro-manager?

Not sure about 2). Maybe you could point me to fix in beta?

**Comment by binli123** [comment]
> For 1), you mean using an MDA that you set up i the GUI in micro-manager, or one that you run from pycro-manager?
> 
> Not sure about 2). Maybe you could point me to fix in beta?

1) The one in the GUI in micro-manager. I haven't tested it in pycro-manager MDA (it wasn't available for color camera), but if they are separate most likely should be okay.
2) It seems to be in this commit https://github.com/micro-manager/micro-manager/commit/88ab1ff051b548aafced024c46f902720eb9d553 , not sure if the same issue for gamma.

**Comment by henrypinkard** [comment]
>     1. The one in the GUI in micro-manager. I haven't tested it in pycro-manager MDA (it wasn't available for color camera), but if they are separate most likely should be okay.

Yes, they are separate systems. I think I got the color channels right but its possible I messed it up since I dont have a real camera to test on.

>     2. It seems to be in this commit [micro-manager/micro-manager@88ab1ff](https://github.com/micro-manager/micro-manager/commit/88ab1ff051b548aafced024c46f902720eb9d553) , not sure if the same issue for gamma.

So I think in this case FIJI stitching is unlikely to work. The reason is that the Micro-manager Tiffs are saved as OME Tiffs, along with a special hack that makes them readable by ImageJ. However, Pycro-manager uses [NDTiffStorage](https://github.com/micro-manager/NDTiffStorage), which is a similar format, but cannot support the IJ/OME metadata because NDTiffStorage allows for much greater flexibility in dataset types than these specifications are designed for. So probably better to use python readers as provided [here](https://pycro-manager.readthedocs.io/en/latest/read_data.html#opening-data-as-dask-array) for stitching. If you're acquiring large tiled datasets, Micro-magellan could be very helpful, and writes to the same format. Caveat: it's possible this reading code needs to be updated for RGB. I can take a look at that
